### ISSUE TITLE
No support for streaming request bodies

### ISSUE THREAD

**Issue by mfine** [libevent/libevent#483]
As a client, I'd like to make a `PUT` with a streaming request body. In `evhttp_make_header_request`, a `Content-Length` for a `PUT` or `POST` that does not have a `Content-Length` will be specified:

```c
	/* Add the content length on a post or put request if missing */
	if ((req->type == EVHTTP_REQ_POST || req->type == EVHTTP_REQ_PUT) &&
	    evhttp_find_header(req->output_headers, "Content-Length") == NULL){
		char size[22];
		evutil_snprintf(size, sizeof(size), EV_SIZE_FMT,
		    EV_SIZE_ARG(evbuffer_get_length(req->output_buffer)));
		evhttp_add_header(req->output_headers, "Content-Length", size);
	}
```

Support for streaming seems well supported on the receiving side for the client. Just no support going from the client to the server. Apologies if I'm misreading the code.

**Comment by azat** [comment]
Hi @mfine 

No you don't misread code, it is not supported for now (and I'm not sure that it will be implemented due to #261, though this is about server part).
As a solution maybe you can send data in chunks?

**Comment by mfine** [comment]
Thanks for the response @azat. Seems like #261 might be pending for some time per https://github.com/ellzey/libevhtp/commit/e09cde823dd346f95d3a4ccaecd8bfb536d65a90. Sorry to hear.
